### ISSUE TITLE
Too much escaping of backslashes from inline Python?

### ISSUE THREAD

**Issue by winniehell** [quarto-dev/quarto-cli#11291]
### Bug description

when printing backslashes from inline Python they get escaped too often

### Steps to reproduce

````qmd
---
title: "Empty"
engine: jupyter
---

Backslash: `{python} "a\\b"` `{python} r"c\\d"`
````

### Expected behavior

Expected output:

```
Backslash: a\b c\\d
```

### Actual behavior

Actual output:

```
Backslash: a\\\\b c\\\\\\\\d
```

### Your environment

$ nixos-version 
24.05.20241012.a3f9ad6 (Uakari)

### Quarto check output

<details>
<summary>quarto check</summary>

```
Quarto 99.9.9
[✓] Checking environment information...
      Quarto cache location: ~/.cache/quarto
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.4.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.46.3: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 99.9.9
      commit: 80bf6b5848bf644de3a4d59e26b5eb067375986a
      Path: ~/quarto-cli/package/dist/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /run/current-system/sw/bin
      Version: undefined

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.5
      Path: ~/quarto-playground/.venv/bin/python3
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```
</details>

**Comment by winniehell** [comment]
workaround:

````qmd
```{python}
from IPython.display import Markdown
```

`{python} Markdown(r"e\\f")`
````

**Comment by cderv** [comment]
Using `Markdown()` is not a workaround but the expected behavior. We document it at https://quarto.org/docs/computations/inline-code.html#markdown-output

> By default, the output of inline expressions is treated as ordinary text (i.e. markdown within it is not rendered). Any markdown-like syntax within the output of inline expressions will be automatically escaped.

Using `Markdown()` is the way to avoid this escaping. 

Hope it helps understand.

**Comment by winniehell** [comment]
@cderv no, it does not help: the Python expression `"a\\b"` rendered as ordinary text is just `a\b`. if that is automatically escaped prior to insertion in Markdown, that would result in `a\\b`. after rendering as PDF or HTML, that should again become `a\b` (and not `a\\\\b`).

while this is a minor thing (because there is an easy way around it), I still think this is unexpected behavior.

**Comment by cderv** [comment]
Thanks for the added precision; Let's look a little closer to this then... 


Let's add `keep-md: true` and also adding block execution so see Python results

````markdown
---
title: "Empty"
engine: jupyter
keep-md: true
---

Backslash: `{python} "a\\b"` `{python} r"c\\d"`

```{python}
"a\\b"
```

```{python}
r"c\\d"
```

````

In the intermediate md we get for the inline

````
Backslash: a\\\\\\\\b c\\\\\\\\\\\\\\\\d
```` 

and for the block

````
```
'a\\b'
```
````

````
```
'c\\\\d'
```
````

This is to produce 
![Image](https://github.com/user-attachments/assets/211be40b-dacf-4b83-8dd2-b4bfaf2c91b8)

As the doc says, "Any markdown-like syntax within the output of inline expressions will be automatically escaped." 
and example give is 
````
`{python} '**not bold**'`
````

which output `\*\*not bold\*\*` leading to `**bold**` in the output. 

So we would indeed expect `a\\b` to be `a\\b` in the output, and not `a\\\\b` - too much escaped string here.

I think I understand your report now. Is that it ?

**Comment by winniehell** [comment]
no worries! thank you for the quick response and for guiding me through the processing 🙂

**Comment by cderv** [comment]
I believe this comes from 
https://github.com/quarto-dev/quarto-cli/blob/4118f40995483ed9e5036c60fdb6c7e66163aa6a/src/core/jupyter/jupyter-inline.ts#L35

where we do indeed probably do two times the `\\` escaping 🤔

**Comment by cscheid** [comment]
This feels quite regression prone, so I'm calling it too late for a 1.6 fix.

**Comment by cderv** [comment]
OK sure. I have a branch already for this as I looked to understand. I'll share for 1.7

**Comment by cderv** [comment]
Let's do this really in 1.8 as this is again too late for 1.7 IMO
